content/en/docs/ops/diagnostic-tools/istioctl-analyze/test.sh

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env bash
+# shellcheck disable=SC1090,SC2154
+
+# Copyright Istio Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e  # Exit on failure
+set -u  # Unset is an error
+# There is no need to echo, output appears in ops_diagnostic-tools_istioctl-analyze_test_debug.txt
+set -o pipefail
+
+# Use 'at' setup profile=default to install default profile?
+# This script doesn't need a control plane, but we need base CRDs
+# @setup profile=none
+
+# This command is allowed to fail
+snip_analyze_all_namespaces || true
+
+# The following step is 'usually' done by cleanup.  Do it here, just in case
+kubectl delete vs ratings || true
+
+snip_analyze_sample_destrule
+
+# TODO Check the output on failure for 'Referenced host+subset in destinationrule not found: "details+v2"' 
+_verify_failure snip_analyze_networking_directory
+
+snip_analyze_all_networking_yaml
+
+snip_analyze_all_networking_yaml_no_kube
```

Comment:
It might be a good idea to put something in the cluster that shows the difference between these two commands, one has an error/warning, while the self contained one doesn't.
```suggestion
# TODO: put some config in the cluster to show a different output between the following two commands.

_verify_contains snip_analyze_all_networking_yaml "No validation issues found"

_verify_contains snip_analyze_all_networking_yaml_no_kube  "No validation issues found"
```

Reply:
Your comment makes sense from a testing standpoint.  From the standpoint of documentation, the point is to show that the `--use-kube` exists, not that it works.
(I could deploy _httpbin_, and show that the analysis of _samples/httpbin/httpbin-gateway.yaml_ is different if we look at httpbin on the cluster.  I am not saying the test won't work, just that it should be tested in integration rather than doc tests.)

Reply:
OK, that's fair enough.